src/PerformanceTests/Distribution.V5.MsmqDistributor.V5/DistributorProfile.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using NServiceBus;
+using Tests.Permutations;
+using Variables;
+
+namespace Distribution.V5.MsmqDistributor.V5
+{
+    using NServiceBus.Config;
+    using NServiceBus.Config.ConfigurationSource;
+
+    public class DistributorProfile : IProfile, INeedPermutation, IProvideConfiguration<MasterNodeConfig>
+    {
+        const string DistributorKey = "Distributor";
+        string node;
+        public Permutation Permutation { private get; set; }
+
+        public void Configure(BusConfiguration cfg)
+        {
+            if (Permutation.ScaleOut != ScaleOut.MsmqDistributor) return;
+
+            var value = this.FetchSetting(DistributorKey);
+
+            if (value == null) throw new InvalidOperationException($"Expected setting {DistributorKey}");
+
+            var isDistributor = string.Equals("distributor", value, StringComparison.InvariantCultureIgnoreCase);
+            var isMaster = string.Equals("master", value, StringComparison.InvariantCultureIgnoreCase);
+
+            if (isDistributor || isMaster)
+            {
+                cfg.RunMSMQDistributor(withWorker: isMaster);
+                return;
+            }
+
+            /*
+            <appSettings>
```

Comment:
@WilliamBZA The config has some settings that you cannot set in code. I left it as reference but if you like I can remove it.